Sources/SourceKitLSP/Rename.swift

```diff
@@ -284,78 +291,438 @@ private extension RenameLocation.Usage {
   }
 }
 
+// MARK: - Name translation
+
+extension SwiftLanguageServer {
+  enum NameTranslationError: Error, CustomStringConvertible {
+    case cannotComputeOffset(Position)
+    case malformedSwiftToClangTranslateNameResponse(SKDResponseDictionary)
+    case malformedClangToSwiftTranslateNameResponse(SKDResponseDictionary)
```

Comment:
I just thought that we pass in the `SKDResponseDictionary` because that gives us more type safety about what the associated type is than when it’s `String`. But I don’t have strong opinions either way.

Reply:
I'd prefer `String`, only so that we don't keep the `SKDResponseDictionary` around. But 🤷 doesn't really matter either way.

Reply:
It won’t stay around for long anyway, let’s just keep it.